core/beacon/src/chain.rs

```diff
@@ -269,4 +271,43 @@ mod tests {
         assert_eq!(other_bc.best_block().header.index, 1);
         assert_eq!(other_bc.get_block(&BlockId::Hash(block1.hash())).unwrap(), block1);
     }
+
+    fn test_fork_choice_rule_helper(graph: Vec<(u32,u32,u32)>, expect: u32) {
+        let storage = Arc::new(create_memory_db());
+        let chain_config = ChainConfig {
+            extra_col: storage::COL_BEACON_EXTRA,
+            header_col: storage::COL_BEACON_HEADERS,
+            block_col: storage::COL_BEACON_BLOCKS,
+            index_col: storage::COL_BEACON_INDEX,
+        };
+        let genesis = BeaconBlock::new(0, CryptoHash::default(), MerkleHash::default(), vec![]);
+        let bc = BlockChain::new(chain_config.clone(), genesis.clone(), storage.clone());
+        let mut blocks: HashMap<u32, CryptoHash> = HashMap::new();
+        blocks.insert(0, genesis.hash());
+        let signer = InMemorySigner::new();
+
+        for (self_id, parent_id, signers) in graph.iter() {
+            let parent = bc.get_block_header_by_hash(blocks.get(parent_id).unwrap()).unwrap();
```

Comment:
`bc.get_header(BlockId::Number(parent_id))`

Reply:
parent_id is not a BlockId, because I intentionally produce multiple blocks with the same height.